app/src/main/java/bankingapp/DBTransactions.java

```diff
@@ -1,140 +1,154 @@
 package bankingapp;
 
 import org.sqlite.SQLiteDataSource;
-
 import java.sql.*;
-import java.util.Scanner;
 
-public class DBTrans {
-    static Scanner scan;
+public class DBTransactions implements IDBTransactions {
+
     static SQLiteDataSource dataSource = new SQLiteDataSource();
     String url;
     Connection connection;
 
-    public DBTrans() throws SQLException{
-        scan = new Scanner(System.in);
-        url = "jdbc:sqlite:account.db";
-        connection = DriverManager.getConnection(url);
+    public DBTransactions() throws SQLException{
+        this.url = "jdbc:sqlite:account.db";
+        this.connection = DriverManager.getConnection(url);
```

Comment:
We're creating a closable resource here (a Connection), but we're not closing it anywhere. DBTransaction should implement `AutoClosable`, so it can clean up its Connection after it's done with it.

In general, if you can't use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) around a closable resource, you should implement AutoClosable instead. The same applies any place where you _new_ an instance of DBTransactions, too!

Reply:
This will be resolved by issue #30